src/neptune/versionHelper.d

```diff
@@ -0,0 +1,334 @@
+
+        return nullAs0(this.patch) - nullAs0(v.patch);
+    }
+
+    /***************************************************************************
+
+        Returns:
+            string representation of this version structure
+
+    ***************************************************************************/
+
+    string toString ( ) const
+    {
+        import std.format;
+        import std.conv;
+
+        return format("v%s.%s.%s%s",
+                      this.major,
+                      this.minor.isNull ? "x" : this.minor.to!string,
+                      this.patch.isNull ? "x" : this.patch.to!string,
+                      this.extra.length > 0 ? "-" ~ this.extra : "");
+    }
+}
+
+
+/// Tests extracting of version information from strings
+unittest
+{
+    alias V = Version;
+    alias T = Version.Type;
+
+    enum V[string] versions = [
+        "v1.5.3" : V(V.Type.Patch, 1, 5, 3),
+        "v1.5.x" : V(V.Type.Minor, 1, 5),
+        "v3.0.x" : V(V.Type.Minor, 3, 0),
+        "v1.x.x" : V(V.Type.Major, 1),
+        "v1.5.0" : V(V.Type.Minor, 1, 5, 0),
+        "v1.0.0" : V(V.Type.Major, 1, 0, 0),
+        "v1.0.1" : V(V.Type.Patch, 1, 0, 1),
+        "v1.1.0" : V(V.Type.Minor, 1, 1, 0),
+        "v1.5.3" : V(V.Type.Patch, 1, 5, 3),
+        "v1.5.3-preview" : V(V.Type.Patch, 1, 5, 3, "preview"),
+        ];
+
+    foreach (name, ver; versions)
+    {
+        auto v = Version(name);
+
+        scope(failure)
+        {
+            import std.stdio;
+            writefln("Failed: %s(%s) != %s(%s)", v, v.type, ver, ver.type);
+        }
+
+        assert(v == ver);
+    }
+}
+
+
+/// Tests sorting of versions
+unittest
+{
+    alias V = Version;
+
+    enum list = [V("v1.1.0"), V("v1.0.1"), V("v3.0.1"), V("v1.0.0"),
+                 V("v3.0"),   V("v1.1.1"), V("v2.0.1"), V("v2.0.0")];
+    enum list_sorted  =
+                [V("v1.0.0"), V("v1.0.1"), V("v1.1.0"), V("v1.1.1"),
+                 V("v2.0.0"), V("v2.0.1"), V("v3.0"),   V("v3.0.1")];
+
+    import std.range;
+    import std.algorithm;
+
+    auto list_mutated = sort(list.dup);
+
+    assert(equal(list_sorted, list_mutated));
+}
+
+
+/*******************************************************************************
+
+    Checks if the current state suggests a patch release.
+
+    Uses the existing tags and the current branch.
+
+    Params:
+        A = result type of a search for a matching major version
+        B = result type of a search for a matching minor version
+        matching_major = result of a search for a matching major version
+        matching_minor = result of a search for a matching minor version
+        current = currently checked out branch
+
+    Returns:
+        a version with type set to Invalid if we guess it will not be a patch
+        release
+
+*******************************************************************************/
+
+public Version needPatchRelease ( A, B ) ( A matching_major, B matching_minor,
```

Comment:
Yeah it looks like this module should be really split in two - smaller part for version management of tags is already implemented via https://github.com/sociomantic-tsunami/neptune/blob/master/src/semver/Version.d (may need few extra features from here but will do for the most part). Branch management and release decision logic is entirely separate domain that should be built on top of semver concept but not be part of it.